components/gantry/oneaxis/oneaxis.go

```diff
@@ -174,57 +147,47 @@ func newOneAxis(ctx context.Context, deps resource.Dependencies, conf resource.C
 		PinEnable := *newConf.LimitPinEnabled
 		oAx.limitHigh = PinEnable
 
-	case 2:
-		oAx.limitType = limitTwoPin
-		board, err := board.FromDependencies(deps, newConf.Board)
-		if err != nil {
-			return nil, err
+		if oAx.mmPerRevolution <= 0 && np == 1 {
+			return nil, errors.New("gantry with one limit switch per axis needs a mm_per_length ratio defined")
 		}
-		oAx.board = board
-
-		PinEnable := *newConf.LimitPinEnabled
-		oAx.limitHigh = PinEnable
-
 	case 0:
-		oAx.limitType = limitEncoder
+		// do nothing, validation takes care of all checks already
 	default:
 		np := len(oAx.limitSwitchPins)
 		return nil, errors.Errorf("invalid gantry type: need 1, 2 or 0 pins per axis, have %v pins", np)
 	}
 
-	if err := oAx.Home(ctx); err != nil {
+	if err := oAx.home(ctx, np); err != nil {
 		return nil, err
 	}
 
+	oAx.posRange = oAx.positionLimits[1] - oAx.positionLimits[0]
+
 	return oAx, nil
 }
 
-func (g *oneAxis) Home(ctx context.Context) error {
+func (g *oneAxis) home(ctx context.Context, np int) error {
```

Comment:
We used to have a public `Home` function, and I had assumed that was part of the `Gantry` interface, but it's not! What does a gantry without a `Home` function look like? I'm surprised `Home` could be removed.

I guess this is okay as-is because homing is not in the interface or the GRPC definitions, I'm just really surprised that it's okay.

Reply:
The home is called upon resource construction. A gantry without homing would be you telling it the limits as well as where it is currently. I would rather have this private until we have an API that exposes it, since you can't use this function out of go/ rdk anyway.

I am pushing for more things being unexported unless absolutely necessary to other packages within rdk.